lib/spack/spack/test/concretize.py

```diff
@@ -1798,3 +1798,21 @@ def test_not_reusing_incompatible_os_or_compiler(self):
         concrete_spec = result.specs[0]
         assert concrete_spec.satisfies('%gcc@4.5.0')
         assert concrete_spec.satisfies('os=debian6')
+
+    def test_git_hash_assigned_version_is_preferred(self):
+        hash = 'a' * 40
+        s = Spec('develop-branch-version@%s=develop' % hash)
+        c = s.concretized()
+        assert hash in str(c)
+
+    @pytest.mark.skipif(sys.platform == 'win32',
+                        reason="Not supported on Windows (yet)")
+    def test_git_hash_version_is_equivalent_to_specified_infinity_version(self):
+        if spack.config.get('config:concretizer') == 'original':
+            pytest.skip(
+                'Original concretizer cannot account for git hashes'
+            )
+        hash = 'a' * 40
+        s = Spec('depends-on-develop ^develop-branch-version@%s=develop' % hash)
+        c = s.concretized()
+        assert hash in str(c)
```

Comment:
Can this work with `s.satisfies('@develop')` and `s.satisfies('@10:')`?

Reply:
I'm not sure.  I'll test.  I was thinking this wouldn't apply to version ranges, just concrete versions.  Not sure if that is the right semantics. Spack version semantics make my head spin at times.  Do you think this feature would be appropriate for an open set of versions?